src/main/java/com/google/api/codegen/java/JavaGapicContext.java

```diff
@@ -120,6 +120,13 @@ public String getTypeName(String typeName) {
   }
 
   /**
+   * Returns the resource name from the entity name.
+   */
+  public String getResourceName(String entityName) {
+    return lowerUnderscoreToUpperCamel(lowerCamelToLowerUnderscore(entityName)) + "Name";
```

Comment:
The "_path" component is coming from here:
https://github.com/googleapis/toolkit/blob/master/src/main/java/com/google/api/codegen/config/CollectionPattern.java#L60

I used lower underscore for the _path suffix because that is how the snippets treat that input. Maybe worth enforcing lower underscore here as well (I can submit in another PR)